### PR TITLE
Feature/tbl 039 moving components into cloud foundry

### PR DESCRIPTION
**Pull request by nherriot**

### CODE REVIEW

File: init_db.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+echo "------ Create database tables ------"
+python oauth2server/manage.py migrate --noinput
+ 
+echo "------ create default oauth2 admin user ------"
+echo "from django.contrib.auth.models import User; User.objects.create_superuser('admin', 'admin@onsemail.com', 'admin2017')" | python oauth2server/manage.py shell
+
```
Comment (liamtoozer) [inline init_db.sh:18]: Should this script have comments on why it's there and what it does? Could this be explained in the README?

File: oauth2server/logs/oauth2_development.log [hunk 2]
```diff
@@ -0,0 +1,95 @@
+[2017-05-30 13:05:57] ERROR [proj.urls.<module>:9] main urls is running
+[2017-05-30 13:05:57] WARNING [proj.urls.<module>:12] main urls is running
+[2017-05-30 13:05:57] CRITICAL [proj.urls.<module>:13] main urls is running
+[2017-05-30 13:06:04] ERROR [proj.urls.<module>:9] main urls is running
+[2017-05-30 13:06:04] WARNING [proj.urls.<module>:12] main urls is running
+[2017-05-30 13:06:04] CRITICAL [proj.urls.<module>:13] main urls is running
```
Comment (liamtoozer) [inline oauth2server/logs/oauth2_development.log]: Should this log be pushed if it's being ignored by git now?

File: oauth2server/logs/oauth2_proj.log [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[2017-05-30 13:19:03] ERROR main urls is running
+[2017-05-30 13:19:03] WARNING main urls is running
+[2017-05-30 13:19:03] CRITICAL main urls is running
```
Comment (liamtoozer) [inline oauth2server/logs/oauth2_proj.log]: Should this log be pushed if it's being ignored by git now?

File: oauth2server/proj/settings/default.py [hunk 4]
```diff
@@ -81,4 +81,128 @@
...
+        'py.warnings': {
+            'handlers': ['console', 'development_logfile'],
+        },
+    }
+}
+
+
+
+
+
```
Comment (liamtoozer) [inline oauth2server/proj/settings/default.py]: Should the commented blocks be removed? Not sure what the policy is with commented out code blocks.

File: oauth2server/proj/settings/docker_settings.py [hunk 5]
```diff
@@ -20,14 +24,40 @@
...
+        if decoded_config[key][0]['name'] == 'postgresql':
+            creds = decoded_config[key][0]['credentials']
+            uri = creds['uri']
+            remoteLogger.info("Found postgres uri string in vcap settings")
+            remoteLogger.info("Postgres DATABASE uri: {}".format(uri))
+
+
+
+else:
+    remoteLogger('VCAP_SERVICES NOT found in os.environ using default SQL database')
```
Comment (liamtoozer) [inline oauth2server/proj/settings/docker_settings.py]: Not completely sure how the logging levels work here sorry, but does this need an explicit level when setting the logging message? E.g. 'info' level?

**Review by liamtoozer — COMMENTED** [review]
Looks good! I've just raised a few comments.

File: oauth2server/proj/settings/default.py [hunk 6]
```diff
@@ -81,4 +81,128 @@
...
+            'handlers': ['console', 'remote_logfile'],
+        },
+        # This is our default django handler loggs
+        'django': {
+            'handlers': ['console', 'development_logfile', 'production_logfile'],
+        },
+        # This is our default warnings log
+        'py.warnings': {
+            'handlers': ['console', 'development_logfile'],
+        },
```
Comment (liamtoozer) [inline oauth2server/proj/settings/default.py]: Test

File: oauth2server/proj/settings/default.py [hunk 7]
```diff
@@ -81,4 +81,128 @@
         'rest_framework.authentication.BasicAuthentication',
         'rest_framework.authentication.SessionAuthentication',
     )
-}
\ No newline at end of file
+}
+
+
+# Custom logging to give us unique logging to work on remote containers and normal logging
```
Comment (liamtoozer) [inline oauth2server/proj/settings/default.py:87]: Should the commented blocks be removed? Not sure what the policy is with commented out code blocks.

**Review by liamtoozer — APPROVED** [review]
Changes have been made after discussion of PR comments. Happy to merge with release/0.1.

File: oauth2server/proj/settings/default.py (hunk 7 above)
Comment (nherriot) [reply oauth2server/proj/settings/default.py:87]: OK. Removed commented out code. I've also updated comments on what loggers are in place.

File: init_db.sh (hunk 1 above)
Comment (nherriot) [reply init_db.sh:18]: OK Liam. I've updated this in my latest commit. :-)

File: oauth2server/logs/oauth2_proj.log (hunk 3 above)
Comment (nherriot) [reply oauth2server/logs/oauth2_proj.log]: Good catch. I've removed the logs. Doh!

**Pull request merged**